bitcoin/src/lib.rs

```diff
@@ -13,21 +13,28 @@
 //! See README.md for detailed documentation about development and supported
 //! environments.
 //!
-//! ## Available feature flags
+//! # Available feature flags
 //!
-//! * `std` - the usual dependency on `std` (default).
+//! * `default` - enables `std` and `secp-recovery`.
+//! * `std` - the usual dependency on `std`.
+//! * `serde` - implements `serde`-based serialization and deserialization.
+//! * `secp-lowmemory` - optimizations for low-memory devices.
 //! * `secp-recovery` - enables calculating public key from a signature and message.
+//! * `bitcoinconsensus` - (dependency) enables validation of scripts/transactions by `rust-bitcoinconsensus`.
+//! * `bitcoinconsensus-std` - same as `bitcoinconsensus` but also enables `std`.
+//! * `rand` - makes it more convenient to generate random values.
+//! * `rand-std` - same as `rand` but also enables `std`.
 //! * `base64` - (dependency), enables encoding of PSBTs and message signatures.
-//! * `rand` - (dependency), makes it more convenient to generate random values.
-//! * `serde` - (dependency), implements `serde`-based serialization and
-//!                 deserialization.
-//! * `secp-lowmemory` - optimizations for low-memory devices.
-//! * `bitcoinconsensus-std` - enables `std` in `bitcoinconsensus` and communicates it
-//!                            to this crate so it knows how to implement
-//!                            `std::error::Error`. At this time there's a hack to
-//!                            achieve the same without this feature but it could
-//!                            happen the implementations diverge one day.
 //! * `ordered` - (dependency), adds implementations of `ArbitraryOrdOrd` to some structs.
+//!
+//! # Note on `foo-std` features
+//!
+//! Feature pairs of form `foo`/`foo-std` are additive (i.e. can both be enabled in a build) however
+//! it is expected that users pick one, for std builds use the `foo-std` feature and only use the
+//! plain `foo` feature for no-std builds. All `foo-std` features also enable `std` so you don't
+//! need both. At times there is some trickery involved to keep the features additive so using the
```

Comment:
I don't like the word "trickery". Can we opt for something more technical and less ambiguous?
Maybe "implementation details" or something like that?